src/Microsoft.Performance.Toolkit.Engine/EngineExecutionResults.cs

```diff
@@ -41,10 +46,12 @@ public sealed class RuntimeExecutionResults
         ///     - or -
         ///     <paramref name="repository"/> is <c>null</c>.
         /// </exception>
+        /// TODO: Going to move RuntimeExecutionResults to internal and expose calls via new interface
         public RuntimeExecutionResults(
             ICookedDataRetrieval cookedDataRetrieval,
             IDataExtensionRetrievalFactory retrievalFactory,
-            IDataExtensionRepository repository)
+            IDataExtensionRepository repository,
+            IDictionary<TableDescriptor, ICustomDataProcessor> tableToProcessorMap)
```

Comment:
>tableToProcessorMap [](start = 63, length = 19)

Guard.NotNull and update doc comment for ArgumentNullException

Reply:
This is going to moved to internal in another PR

Reply:
Added Guard.NotNull